cmasher/utils.py

```diff
@@ -233,6 +239,117 @@ def _get_cmap_perceptual_rank(
 
 
 # %% FUNCTIONS
+# This function combines multiple colormaps at given nodes
+def combine_cmaps(
+    cmaps: list[Union[Colormap, str]],
+    *,
+    nodes: Optional[Union[list[float], np.ndarray]] = None,
+    n_rgb_levels: int = 256,
+    combined_cmap_name: str = "combined_cmap",
+) -> LinearSegmentedColormap:
+    """Create a composite matplotlib colormap by combining multiple colormaps.
+
+    Parameters
+    ----------
+        cmaps: list of Colormap or colormap names (str)
+            list of colormaps to combine.
+        nodes: list or numpy array of nodes (float). Defaults: equal divisions.
+            The blending points between colormaps, in the range [0, 1].
+        n_rgb_levels: int. Defaults: 256.
+            Number of RGB levels for each colormap segment.
+        combined_cmap_name: str. Defaults: "combined_cmap".
+            name of the combined Colormap.
+
+    Returns
+    -------
+        Colormap: The composite colormap.
+
+    Raises
+    ------
+        TypeError: If the list contains mixed datatypes or invalid
+            colormap names.
+        ValueError: If the cmaps contain only one single colormap,
+            or if the number of nodes is not one less than the number
+            of colormaps, or if the nodes do not contain incrementing values
+            between 0.0 and 1.0.
+
+    Note
+    ----
+        The colormaps are combined from low value to high value end.
+
+    References
+    ----------
+        - https://stackoverflow.com/questions/31051488/combining-two-matplotlib-colormaps/31052741#31052741
```

Comment:
I don't think linking to SO brings much value here so I'd rather we drop this section
```suggestion
```

Reply:
I'm not 100% about the best pratice here, but for the implementation I referred to this thread so I thought it might be good to quote the original author here for "academic integrity" reason?

Reply:
Why not, I guess !